Change base image and add a package to run FahCore_23

sanori opened the pull request:
Since FahCore 23 (and OpenMM 8.0.3) requires OpenCL 3.0, we need to upgrade CUDA and Ubuntu versions.
- Ubuntu: 18.04 -> 22.04 (OpenCL 3.0 is supported since Ubuntu 22.04)
- CUDA:  11.2.2 -> 11.7.1 (Official CUDA image version that supports Ubuntu 22.04 is 11.7.1)

FahCore 23 requires libexpat1 shared library. So, install libexpat1 package.

This PR resolves Issue #31 .

sanori committed:
feat: change base image and add package to run FahCore_23

manhtruongwang commented:
Amazing fix, thanks you

cgint commented:
I faced the same issue as described in https://github.com/FoldingAtHome/containers/issues/31.

@sanori Thx for your PR 

That fixed it for me and now everything works inside docker the same way as on my host machine.

sanori committed:
chore: fix the URL of fahclient deb file

sanori committed:
chore: revert base image to CUDA 11.2.2

sanori commented:
@beberg
As you mentioned, I've reverted the base image version to nvidia/cuda:11.2.2-base-ubuntu18.04 to maximize compatibility.
Cores 0x23-0x27 appear to depend on `libexpat1`, so this PR should fix the issue.
I've also fixed the URL for `fahclient_7.6.21_amd64.deb`.

Are there any other issues that need to be fixed before merging?

jcoffland commented:
The v7 client is so old now.  I don't want to encourage its use.  It would be much better if this PR used the latest public release of the v8 client.

beberg commented:
The reason this has been sitting is that the huge number of cores developed since 1999 all have different dependencies as they were built over the years. So any change made to the container is a "fix-one break-two" kind of situation. Any change then interacts with the assignment servers, host hardware, host drivers, orchestration settings... you get the idea.

I'm going to do some more work on this in the near future, but for now we don't have a good test harness to run all the cores in one environment to know if a change is helping or hurting overall. We probably need to build that first, and having those tests would help with non-container development as well.
